### PR TITLE
FoundationEssentials: enable unprivilleged symlinks if possible

### PR DESCRIPTION
**Pull request by compnerd**
If developer mode is enabled or the user is assigned the SeCreateSymbolicLinkPrivilege privilege, permit the unelevated symbolic link creation.

### CODE REVIEW

**Review by parkera — COMMENTED** [review]
Is this a behavior most clients would expect, or does it need to be propagated out as some kind of option from Foundation?

**Comment by compnerd** [comment]
I think that anyone coming from the Unix would would expect this. On the Windows world, I think that anything who enabled developer mode would expect this. However, since it would not impact anyone on Windows who does not have developer mode or symlink creation privilege explicitly assigned, it would be pretty expected I would say.

**Comment by compnerd** [comment]
@swift-ci please test macOS platform

**Comment by compnerd** [comment]
@swift-ci please test macOS platform

File: Sources/FoundationEssentials/WinSDK+Extensions.swift [hunk 1]
```diff
@@ -145,6 +145,18 @@ package var PAGE_READONLY: DWORD {
     DWORD(WinSDK.PAGE_READONLY)
 }
 
+package var PATHCCH_ALLOW_LONG_PATHS: ULONG {
+    ULONG(WinSDK.PATHCCH_ALLOW_LONG_PATHS.rawValue)
+}
+
+package var RRF_RT_REG_SZ: DWORD {
```
Comment (jmschonfeld) [inline Sources/FoundationEssentials/WinSDK+Extensions.swift:152]: It looks like this declaration and the one above aren't used in this change. Are they necessary?

Comment (compnerd) [reply Sources/FoundationEssentials/WinSDK+Extensions.swift:152]: They are used in subsequent PRs that I have in my tree. The PRs take a while to get merged and the overloads here are just difficult to keep separated.

**Comment by jmschonfeld** [comment]
@swift-ci please test macOS platform

**Review by jmschonfeld — APPROVED** [review]

**Pull request merged**